### PR TITLE
Fix #449

### LINKED ISSUE #449: $cpp_options() doesn't return options from make/local
If I have compiler flags in make/local, and I compile a model, these flags don't show up in `$cpp_options()`.

For example:
```
library(cmdstanr)
rebuild_cmdstan()
cpp_options <- list(
  "STAN_THREADS=TRUE"
)
cmdstan_make_local(cpp_options = cpp_options)
rebuild_cmdstan()
logistic1 <- cmdstan_model("/Users/JacobSocolar/Desktop/logistic1.stan", cpp_options = list(stan_threads = TRUE))
logistic1_makelocal <- cmdstan_model("/Users/JacobSocolar/Desktop/logistic1.stan")
logistic1$cpp_options()
logistic1_makelocal$cpp_options()
```
It seems like `$cpp_options()` should return the same regardless of whether the option is specified in make/local or `cmdstan_model(... , cpp_options)`.  Likewise, it seems like `$cpp_options()` should return compiler flags like `CXXFLAGS+= -O3 -march=native -mtune=native`, which *cannot* be passed via `cmdstan_model(... , cpp_options)`

**CmdStanR version number** 0.3.0

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

Fixes #449 or at least a major part of it by handling the C++ compiler options when they are supplied via the make/local file.

As that issue states, if you add `STAN_THREADS=true` to the make/local file `mod$cpp_options()` will not return `STAN_THREADS` as `mod$cpp_options()` only handles what is supplied to cmdstan_model or $compile().

#449 suggests handling the entire make/local file, but that is not really doable.

What this PR adds can be best explained with an example:

```r
library(cmdstanr)

backup <- cmdstan_make_local()

cmdstan_make_local(cpp_options = list(), append = FALSE)

file <- file.path(cmdstan_path(), "examples", "bernoulli", "bernoulli.stan")
mod <- cmdstan_model(file)
# should obviously not print anything about stan threads (on master or this PR)
print(mod$cpp_options()) 

file.remove(mod$exe_file())

cmdstan_make_local(cpp_options = list(stan_threads = TRUE))

file <- file.path(cmdstan_path(), "examples", "bernoulli", "bernoulli.stan")
mod <- cmdstan_model(file)
# should print that stan threads are on on this branch, but would not print anything on master
print(mod$cpp_options())

file.remove(mod$exe_file())

# restore
cmdstan_make_local(cpp_options = backup, append = FALSE)
```

Once we also address https://github.com/stan-dev/cmdstanr/issues/488 $cpp_options() will also print Stan C++ options for already created files (provided they were compiled with 2.27+.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)    
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: R/data.R [hunk 1]
```diff
@@ -191,7 +191,7 @@ process_data <- function(data, model_variables = NULL) {
       }
     }
     path <- tempfile(pattern = "standata-", fileext = ".json")
-    write_stan_json(data = data, file = path, always_decimal = (cmdstan_version() > "2.26.1"))
+    write_stan_json(data = data, file = path, always_decimal = !is.null(model_variables))
```
Comment (rok-cesnovar) [inline R/data.R:194]: This is unrelated just a random fix.

**Review by jgabry — COMMENTED** [review]
Thanks! Looks like tests are passing but I haven't had a chance to review yet. I can hopefully review this tomorrow.

**Review by jgabry — APPROVED** [review]
Changes look good to me. Looks like tests are running after merging in master, but I think this is ready to be merged if those pass.
